qiskit_experiments/framework/composite/composite_experiment.py

```diff
@@ -93,13 +92,20 @@ def _initialize_experiment_data(self):
         return experiment_data
 
     def _additional_metadata(self):
-        return {"component_job_metadata": []}
-
-    def _add_job_metadata(self, experiment_data, jobs, **run_options):
-        # Extract component metadata
-        component_metadata = []
+        """Add component experiment metadata"""
+        return {
+            "component_metadata": [sub_exp._metadata() for sub_exp in self.component_experiment()]
+        }
+
+    def _add_job_metadata(self, metadata, jobs, **run_options):
+        super()._add_job_metadata(metadata, jobs, **run_options)
+
+        # Add component job metadata
+        if "component_metadata" not in metadata:
```

Comment:
Probably we can remove this branching and always insert `component_metadata` so that we can remove `_additional_metadata`. Currently we always need to write
```python
def _additional_metadata(self):
    additional_metadata = super()._additional_metadata()
    additional_metadata["child_data"] = ...
    return additional_metadata
``` 
in every subclass (if we want to add experiment-level metadata), but this is bit annoying.

Reply:
Adding that metadata to this job function would be possible, but it is going a little around the intention of these two functions. The additional metadata is adding the sub-experiment metadata for later use, while one is only copying the job for the sub experiments. The conditional check here was just making sure the fields to be updated actually exist but it should be unnecessary since they should always exist so i can remove it.